res/controllers/Reloop-Ready-scripts.js

```diff
@@ -0,0 +1,1141 @@
+                if (undefined !== this.group && this.colorKey !== undefined) {
+                    this.connections[1] = engine.makeConnection(this.group, this.colorKey, function(id) {
+                        if (engine.getValue(this.group, this.outKey)) {
+                            this.outputColor(id);
+                        }
+                    });
+                }
+            };
+            if (this.connections[0] !== undefined) {
+                this.disconnect();
+                this.connect();
+                this.trigger();
+            }
+        },
+    });
+    this.pads = new components.ComponentContainer();
+    for (let n = 0; n <= 7; n++) {
+        this.pads[n] = new this.PerformancePad(n);
+    }
+
+    this.parameterLeft = new components.Button({
+        midi: [0x94 + index, 0x28],
+        shiftOffset: 0x2,
+        input: ReloopReady.makeButtonDownInputHandler(function() {
+            if (range === PitchPlayRange.UP) {
+                range = PitchPlayRange.MID;
+            } else if (range === PitchPlayRange.MID) {
+                range = PitchPlayRange.DOWN;
+            } else {
+                range = PitchPlayRange.UP;
+            }
+            theContainer.forEachComponent(c => c.trigger());
+        }),
+    });
+    this.parameterRight = new components.Button({
+        midi: [0x94 + index, 0x29],
+        shiftOffset: 0x2,
+        input: ReloopReady.makeButtonDownInputHandler(function() {
+            if (range === PitchPlayRange.UP) {
+                range = PitchPlayRange.DOWN;
+            } else if (range === PitchPlayRange.MID) {
+                range = PitchPlayRange.UP;
+            } else {
+                range = PitchPlayRange.MID;
+            }
+            theContainer.forEachComponent(c => c.trigger());
+        }),
+    });
+};
+
+ReloopReady.Pitch.prototype = Object.create(ReloopReady.PadMode.prototype);
+
+
+// There is no such thing as a scratch bank in Mixxx so I'm repurpusing this
+// PadMode for beatjumping.
+ReloopReady.ScratchBankPadMode = function(index) {
+    ReloopReady.PadMode.call(this);
+
+    this.currentJumpSizeExp = -2;
+    const theContainer = this;
+
+    const clampJumpSizeExp = jumpSizeExp => _.clamp(jumpSizeExp, -5, 6);
+
+    this.setjumpSizeExp = jumpSizeExp =>  {
+        const middle = this.pads.length / 2;
+        let jumpsize = Math.pow(2, jumpSizeExp);
+        for (let i = 0; i < middle; ++i) {
+            const padTop = this.pads[i];
+            padTop.inKey = `beatjump_${jumpsize}_forward`;
+            padTop.outKey = padTop.inKey;
+            const padBot = this.pads[i + middle];
+            padBot.inKey = `beatjump_${jumpsize}_backward`;
+            padBot.outKey = padBot.inKey;
+            jumpsize *= 2;
+        }
+    };
+
+
+    for (let i = 0; i < this.pads.length; i++) {
+        this.pads[i] = new components.Button({
+            midi: [0x94 + index, 0x14 + i],
+            group: `[Channel${index + 1}]`,
```

Comment:
What is i and what is index? Better use self descritive names like padIdx and padModeIdx.